cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.plugin;
+
+import co.cask.cdap.api.macro.InvalidMacroException;
+import co.cask.cdap.api.macro.MacroEvaluator;
+
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * This class provides static utility functions for parsing macro syntax including:
+ *    - Maco Substitution
+ *    - Syntax validation
+ *    TODO: - Geting a list of macros a user must configure in a String
+ */
+public class MacroParser {
+  private static final String[] ESCAPED_TOKENS = {"\\${", "\\}", "\\(", "\\)"};
+  private static final Pattern ARGUMENT_DELIMITER = Pattern.compile(",");
+  private static final int MAX_SUBSTITUTION_DEPTH = 10;
+
+  private final MacroEvaluator macroEvaluator;
+
+  public MacroParser(MacroEvaluator macroEvaluator) {
+    this.macroEvaluator = macroEvaluator;
+  }
+
+  /**
+   * Checks if a string contains a well-formatted macro.
+   * @param str the raw string containing macro syntax.
+   * @throws InvalidMacroException on invalid macro syntax.
+   * @return if str contains a macro.
+   */
+  public boolean containsMacro(String str) throws InvalidMacroException {
+    // used to check macro syntax
+    substituteMacro(str, 0, new PropertyMacros(), false, false);
+    return (findRightmostMacro(str, null, false, false) != null);
+  }
+
+  /**
+   * Creates a set of all macro arguments present in a string that a user must specify to expand the string.
+   * TODO: Get this working for REST API
```

Comment:
name it like integrate this for REST API access. it sounds like it wouldn't work with REST API access.